test/cql-pytest/test_describe.py

```diff
@@ -447,6 +447,49 @@ def test_desc_index(cql, test_keyspace):
         
         assert f"CREATE INDEX named_index ON {tbl}{maybe_space}(c)" in c_desc
 
+def test_desc_index_on_collections(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int primary key, b int, c frozen<set<int>>, d map<int, int>, e list<int>, f set<int>, g list<int>, h set<int>") as tbl:
+        indexes = [
+            ("idx_b",           f"CREATE INDEX idx_b ON {tbl}(b)"),
+            ("idx_c",           f"CREATE INDEX idx_c ON {tbl}(full(c))"),
+            ("idx_d_entries",   f"CREATE INDEX idx_d_entries ON {tbl}(entries(d))"),
+            ("idx_d_keys",      f"CREATE INDEX idx_d_keys ON {tbl}(keys(d))"),
+            ("idx_d_values",    f"CREATE INDEX idx_d_values ON {tbl}(values(d))"),
+            ("idx_e",           f"CREATE INDEX idx_e ON {tbl}(values(e))"),
+            ("idx_f",           f"CREATE INDEX idx_f ON {tbl}(values(f))"),
```

Comment:
Can we include some cases that would exercise the "unquoting" logic?

Reply:
Cases with unquoting are in another test in this commit (test_desc_index_column_quoting)

Reply:
Oh, right. Sorry, I missed them. Nonetheless, more test cases should be added to cover strings with escaped characters.

Reply:
See also the tests in test/cql-pytest/test_secondary_index.py, test_index_quoted_names, which checks all the interesting edge cases of the quoting (not describe table of course, just the create index).

The most "difficult" edge case is a column name like `keys(m)` which looks like it's a collection, but isn't. This case is handled correctly, and tested by that test and we should also test it in the context of DESCRIBE.

Reply:
copied the cases from the test @nyh  pointed